src/linear_algebra/basis.lean

```diff
@@ -243,6 +243,38 @@ of_repr (f.symm.trans b.repr)
 
 end map
 
+section map_coeffs
+
+variables {R' : Type*} [semiring R'] [module R' M] (f : R ≃+* R') (h : ∀ c (x : M), f c • x = c • x)
+
+include f h
+
+/-- If `R` and `R'` are isomorphic rings that act identically on a module `M`,
+then a basis for `M` as `R`-module is also a basis for `M` as `R'`-module.
+
+See also `basis.algebra_map_coeffs` for the case where `f` is equal to `algebra_map`.
+-/
+@[simps {simp_rhs := tt}]
+def map_coeffs : basis ι R' M :=
+of_repr { map_smul' := λ c x,
+  begin
+    ext i,
+    rw [← f.apply_symm_apply c, h],
+    simp only [finsupp.map_range.add_equiv_apply, linear_equiv.coe_to_add_equiv, finsupp.coe_smul,
+              add_equiv.coe_trans, function.comp_app, add_equiv.to_fun_eq_coe, smul_eq_mul,
+              finsupp.map_range_apply, pi.smul_apply, ring_equiv.to_add_equiv_eq_coe,
+              ring_equiv.coe_to_add_equiv, linear_equiv.map_smul, ring_equiv.map_mul]
+  end
+  .. b.repr.to_add_equiv.trans (finsupp.map_range.add_equiv f.to_add_equiv) }
```

Comment:
After merging #8701, I could only manage to shave off one line from the alternate proof, not so impressive. If you don't mind, I'll stick with the original.
```suggestion
include b
/-- If `R` and `R'` are isomorphic rings that act identically on a module `M`,
then a basis for `M` as `R`-module is also a basis for `M` as `R'`-module. -/
@[simps {simp_rhs := tt}]
def map_coeffs : basis ι R' M :=
let inst : module R' R := module.comp_hom R (↑f.symm : R' →+* R) in by exactI (
have sc : is_scalar_tower R' R M := { smul_assoc := λ x y z, begin
  dsimp [(•)],
  rw [mul_smul, ←h, f.apply_symm_apply],
end },
by exactI (of_repr $ (b.repr.restrict_scalars R').trans $ finsupp.map_range.linear_equiv begin
  refine {map_smul' := λ r' r, _, .. (↑f : _ ≃+ _)},
  apply f.symm.injective,
  simp,
  refl,
end))
```

Reply:
Here's it in fewer:
```suggestion
include b

@[simps {simp_rhs := tt}]
def map_coeffs : basis ι R' M :=
let inst : module R' R := module.comp_hom R (↑f.symm : R' →+* R) in by exactI (
have sc : is_scalar_tower R' R M := { smul_assoc := λ x y z, begin
  dsimp [(•)],
  rw [mul_smul, ←h, f.apply_symm_apply],
end },
by exactI (of_repr $ (b.repr.restrict_scalars R').trans $
  finsupp.map_range.linear_equiv (module.comp_hom.to_linear_equiv f.symm).symm ))
```

Reply:
Or compressed to
```lean
@[simps {simp_rhs := tt}]
def map_coeffs : basis ι R' M :=
begin
  letI : module R' R := module.comp_hom R (↑f.symm : R' →+* R),
  haveI : is_scalar_tower R' R M :=
  { smul_assoc := λ x y z, begin dsimp [(•)],  rw [mul_smul, ←h, f.apply_symm_apply], end },
  exact (of_repr $ (b.repr.restrict_scalars R').trans $
    finsupp.map_range.linear_equiv (module.comp_hom.to_linear_equiv f.symm).symm )
end
```